lib/src/bits.h

```diff
@@ -0,0 +1,29 @@
+#ifndef TREE_SITTER_BITS_H_
+#define TREE_SITTER_BITS_H_
+
+#include <stdint.h>
+
+static inline uint32_t bitmask_for_index(uint16_t id) {
+  return (1u << (31 - id));
+}
+
+#ifdef _WIN32
+
+#include <intrin.h>
+
+static inline uint32_t count_leading_zeros(uint32_t x) {
+  if (x == 0) return 32;
+  uint32_t result;
+  _BitScanReverse(&result, x);
+  return 31 - result;
+}
+
+#else
+
+static inline uint32_t count_leading_zeros(uint32_t x) {
+  if (x == 0) return 32;
+  return __builtin_clz(x);
+}
```

Comment:
When matching queries, we need to keep track of multiple *lists* of captured nodes. I use an object pool data structure called `CaptureListPool` to manage these lists. The `CaptureListPool` uses a bitmap to keep track of which lists are in use. I find the first free list in the bitmap using this `count_leading_zeros`, which maps to different intrinsic functions on POSIX and windows.

In the WASM world, I wasn't sure if emscripten was going to support `__builtin_clz`, but it does!